src/main/java/net/minecraftforge/client/event/EntityViewRenderEvent.java

```diff
@@ -83,23 +90,67 @@ public void setDensity(float density)
     }
 
     /**
-     * Event that allows any feature to customize the rendering of fog.
+     * This event allows for customization of parameters related to fog rendering. The plane distances are based on the player's render distance.
+     *
+     * A FogMode of FOG_SKY is the skybox's fog.
+     * A FogMode of FOG_TERRAIN is what the fog induced by render distance uses. This works best for reducing the camera's visibility.
      */
-    @HasResult
-    public static class RenderFogEvent extends FogEvent
+    public static class RenderFogEvent extends EntityViewRenderEvent
     {
+        private final FogMode type;
         private final float farPlaneDistance;
+        private final float nearPlaneDistance;
+        private final FogShape fogShape;
 
+        @Deprecated //TODO: remove in 1.19
         public RenderFogEvent(FogMode type, Camera camera, float partialTicks, float distance)
         {
-            super(type, camera, partialTicks);
-            this.farPlaneDistance = distance;
+            this(type, camera, partialTicks, 0f, distance, FogShape.SPHERE);
```

Comment:
Near plane is treated as `-8f` as default in the hook so that probably should be reflected here.